src/public/script.js

```diff
@@ -4,140 +4,133 @@ const limitSelect = document.querySelector('#limit-select');
-        }, 10 * 1000);
+    const requestsToSend = parseInt(limitSelect.value);
+    const whenToSendTick = Math.ceil(100 / requestsToSend);
 
-        timerDiv.innerHTML = `${counter} seconds left`;
-
-        interval = setInterval(() => {
-            counter--;
-            timerDiv.innerHTML = `${counter} seconds left`;
-        }, 1000);
-    }
-
-    let results = [];
-    let calls = [];
+    sendButton.disabled = true;
+    limitSelect.disabled = true;
 
     const callPing = async () => {
         try {
-            const response = await axios.get(pingUrl, {
+            await axios.get(pingUrl, {
                 headers: {
                     'Cache-Control': 'no-cache, no-store, must-revalidate',
                     Pragma: 'no-cache',
                     Expires: '0'
                 }
             });
 
-            results.push(response);
+            successfullRequests++;
         } catch (err) {
-            results.push(err.response);
+            blockedRequests++;
         }
     };
 
-    for (let i = 0; i < 2; i++) {
-        calls.push(callPing());
-    }
+    timerDiv.innerHTML = `${counter} seconds left`;
 
-    await Promise.all(calls);
+    let result = document.createElement('p');
 
-    results.forEach(result => {
-        if (!result || !result.status || (result.status !== 200 && result.status !== 429)) {
-            responses.unknown++;
-        }
-        if (result.status === 200) {
-            responses.successful++;
-        }
-        if (result.status === 429) {
-            responses.limited++;
-        }
-    });
+    result.classList.add('lead');
+    result.id = `result-${resultAmount}`;
 
-    resultDiv.innerHTML = '';
+    const sentMessage = document.createElement('span');
+
+    sentMessage.classList.add('sent');
+
+    const successfullMessage = document.createElement('span');
+
+    successfullMessage.classList.add('successfull');
+    successfullMessage.style.color = 'green';
+
+    const blockedMessage = document.createElement('span');
+
+    blockedMessage.classList.add('blocked');
+    blockedMessage.style.color = 'red';
+
+    result.appendChild(sentMessage);
+    result.appendChild(successfullMessage);
+    result.appendChild(blockedMessage);
+
+    resultDiv.prepend(result);
 
-    if (responses.successful) {
-        const successfullMessage = document.createElement('h3');
+    requestInterval = setInterval(async () => {
+        if (tick % whenToSendTick === 0) {
+            await callPing();
 
-        successfullMessage.style.color = 'green';
-        successfullMessage.innerHTML = `${responses.successful} requests were handled properly (status 200)`;
-        resultDiv.appendChild(successfullMessage);
-    }
+            requestsSentCount++;
 
-    if (responses.limited) {
-        const limitedMessage = document.createElement('h3');
+            const currentResultsSent = document.querySelector(`#result-${resultAmount} .sent`);
```

Comment:
cannot be this moved outside interval function?

Reply:
it turned out that it's not needed at all, I can use previously created element and we don't need to assign classes/id to messages :)